### DISCUSSION TITLE
Nightscout self hosted does not receive treatments from AAPS. NSClient error: io.socket.engineio.client.EngineIOException: xhr poll error

### DISCUSSION THREAD

**Discussion by TiagoPRSilva** [NightscoutFoundation/xDrip#2776] [General]
Hello i have nightscout self hosted in docker and behind an nginx reverse proxy (swag docker container). Xdrip is able to upload readings to nightscout. As for treatments, AAPS keeps getting 'NSClient error: io.socket.engineio.client.EngineIOException: xhr poll error'.
After some analysis i found these logs in nginx After all i found some relevant logs in nginx.

`tail -f access.log  my-redacted-public-ip - - [11/Apr/2023:22:49:22 +0100] "GET /api/v1/treatments HTTP/2.0" 200 2 "-" "okhttp/3.12.13" my-redacted-public-ip - - [11/Apr/2023:22:49:22 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:49:23 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:49:25 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:49:27 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:49:32 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:49:37 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:49:43 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:49:48 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:49:53 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:49:58 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:50:03 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:50:08 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:50:13 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:50:18 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:50:23 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:50:27 +0100] "POST /api/v1/entries HTTP/2.0" 200 68 "-" "okhttp/3.12.13" my-redacted-public-ip - - [11/Apr/2023:22:50:27 +0100] "POST /api/v1/devicestatus HTTP/2.0" 200 161 "-" "okhttp/3.12.13" my-redacted-public-ip - - [11/Apr/2023:22:50:27 +0100] "GET /api/v1/treatments HTTP/2.0" 200 2 "-" "okhttp/3.12.13" my-redacted-public-ip - - [11/Apr/2023:22:50:28 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:50:33 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:50:38 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:50:43 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0" my-redacted-public-ip - - [11/Apr/2023:22:50:48 +0100] "GET /socket.io/?EIO=3&transport=polling HTTP/2.0" 400 51 "-" "okhttp/4.9.0"`

Is there anyone with a similar setup (self hosted) that had similar issues and found a solution ?

**Comment by Navid200** [comment]
If I understand correctly:

1- You have set up Nightscout not using our (xDrip) instructions https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html.
You have either set it up your way or following someone else's instructions.

2- xDrip is uploading successfully to your Nightscout.

3- AAPS cannot upload to your Nightscout.


If those are all correct, it seems to me the best place to start looking for answers is AAPS and the group whose instructions you used to set up your Nightscout.

**Comment by swissalpine** [comment]
Probably the proxy server is not configured correctly. In the following issue thread you can find hints:
https://github.com/nightscout/AndroidAPS/issues/75

**Reply by misiek2lucky** [reply]
Thanks man. I forgot set websocket on my proxy. Thanks for remind me.
